app/controllers/paypal_accounts_controller.rb

```diff
@@ -84,7 +92,7 @@ def create
     when Some(:verified)
       create_billing_agreement
     else
-      create_paypal_account
+      redirect_to action: ask_order_permission
```

Comment:
Similarly here we have inconsistent abstraction level in the method making it hard to understand. The above branch delegates control and outcome to a helper method but else branch deals with sending back a redirect. Again, my suggestion would be to handle rendering / redirecting here and use helpers to deal with pp api if needed.